pex/pip/version.py

```diff
@@ -25,29 +39,24 @@ def __init__(
         # type: (...) -> None
         super(PipVersionValue, self).__init__(version)
 
-        def to_requirement(
-            project_name,  # type: str
-            project_version=None,  # type: Optional[str]
-        ):
-            # type: (...) -> str
-            return (
-                "{project_name}=={project_version}".format(
-                    project_name=project_name, project_version=project_version
-                )
-                if project_version
-                else project_name
-            )
-
-        self.requirement = requirement or to_requirement("pip", version)
-        self.setuptools_requirement = to_requirement("setuptools", setuptools_version)
-        self.wheel_requirement = to_requirement("wheel", wheel_version)
+        self.requirement = requirement or _to_requirement("pip", version)
+        self.setuptools_requirement = _to_requirement("setuptools", setuptools_version)
+        self.wheel_requirement = _to_requirement("wheel", wheel_version)
         self.requires_python = SpecifierSet(requires_python) if requires_python else None
 
     @property
     def requirements(self):
         # type: () -> Iterable[str]
         return self.requirement, self.setuptools_requirement, self.wheel_requirement
 
+    def override_setuptools_version(self, version):
```

Comment:
Adding methods to allow mutating these previously-immutable (at least as immutable as Python lets you get) enum values makes my skin crawl a bit, but it was the "cleanest" way I could figure out how to expose the overrides. My other ideas involved changing the constructor for `PipVersion` or `PipVersionValue`, which would have sprawled to every usage of a `PipVersion.xyz` in the codebase...

Reply:
Agreed - this is not awesome. It seems like the right thing to do is separate setuptools and wheel versions from the PipVersion enum; so you have just the PipVersion enum, then an external default mapping of those to setuptools and wheel versions and the new flag overrides that external default mapping.

Reply:
👍 I'll rework